mattermost/mattermost
# [MM-53650] Add disable emoticon rendering setting to webapp

#### Summary


This pull request adds a user setting to the webapp to toggle rendering emoticons (`:D`) as emojis (😄).

The setting is added as a component in `components/user_settings/display/render_emoticons_as_emoji/` which is imported in `components/user_settings/display/user_settings_display.tsx`.

The setting is stored as a user preference using the `savePreferences()` action.

I've added constants for the preference to `utils/constants.tsx` and `webapp/channels/src/packages/mattermost-redux/src/constants/preferences.ts`.

To actually use the setting, I've modified `components/post_markdown` to receive it's value as a prop, for which I've used `getBool()`. `post_markdown` passes this value down to `Markdown` on the `options` object, which then passes it down to `utils/text_formatting.tsx`, which finally passes the value to `emoticons.tsx` as a newly added parameter. `emoticons.tsx` checks whether the value is true and if it is, it transforms the emoticons into emojis.

I've updated affected tests and created unit tests for the new component. I've also updated the English translation file.

##### QA Test Steps

1. Navigate to User Settings.
2. Go to the Display category.
3. Find the section labelled "Auto-render emoticons as emoji" and click "Edit".
4. Toggle the setting and click "Save".
5. Emoticon rendering on messages sent by the current user and other users should be toggled client-side with the setting.

#### Ticket Link


Fixes (partially) https://github.com/mattermost/mattermost/issues/26504
Jira https://mattermost.atlassian.net/browse/MM-53650

**Note** the issue and ticket describe adding this feature to the mobile app as well, which this PR does not.

#### Screenshots


|  before  |  after  |
|----|----|
| ![image](https://github.com/user-attachments/assets/17600204-6446-4206-ab86-a8aea7ed05b3) | ![image](https://github.com/user-attachments/assets/cc7ac7bf-1eb8-47c5-bb93-abce42bc006a) |

#### Demo


https://github.com/user-attachments/assets/6e034b02-85f0-4ac1-b27d-4f945e1acd07


#### Release Note

```release-note
Added a new user setting to toggle rendering emoticons (:D) as emojis (😄)
```
